tuf/utils/x509.go

```diff
@@ -443,6 +454,9 @@ func EncryptPrivateKey(key data.PrivateKey, role, passphrase string) ([]byte, er
 		return nil, fmt.Errorf("unable to encrypt key - invalid PEM file produced")
 	}
 	encryptedPEMBlock.Headers["role"] = role
+	if gun != "" {
```

Comment:
Agreed that we don't want PEM headers with empty values.  Should we do the same for the `"role"` value then?

I've traced through the code and it seems like this might be a condition on a passphrase change for imported delegation keys since that calls `AddKey` on a key that could potentially have no role, and it would be nice to keep this consistent between GUNs and roles.  I can envision delegation keys in particular not needing a role in the PEM headers, since a key could be used across multiple delegation roles.